go/enclave/rollupchain/rollup_chain.go

```diff
@@ -657,27 +616,54 @@ func (rc *RollupChain) validateRollup(rollup *core.Rollup, rootHash common.L2Roo
 }
 
 // Calculates the state after processing the provided block.
-func (rc *RollupChain) handlePostGenesisBlock(block *types.Block) (*common.L2RootHash, error) {
+func (rc *RollupChain) handlePostGenesisBlock(block *types.Block) (*common.L2RootHash, *core.Rollup, error) {
 	// TODO - #718 - Cannot assume that the most recent rollup is on the previous block anymore. May be on the same block.
-	currentHeadRollupHash, err := rc.storage.FetchHeadRollupForL1Block(block.ParentHash())
+	// We retrieve the current L2 head.
+	l2HeadHash, err := rc.storage.FetchHeadRollupForL1Block(block.ParentHash())
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not retrieve current head rollup hash. Cause: %w", err)
+	}
+	l2Head, err := rc.storage.FetchRollup(*l2HeadHash)
 	if err != nil {
-		return nil, fmt.Errorf("could not retrieve current head rollup hash. Cause: %w", err)
+		return nil, nil, fmt.Errorf("could not fetch parent rollup. Cause: %w", err)
 	}
-	currentHeadRollup, err := rc.storage.FetchRollup(*currentHeadRollupHash)
+	rollupTxReceipts, err := rc.storage.GetReceiptsByHash(*l2Head.Hash())
 	if err != nil {
-		return nil, fmt.Errorf("could not fetch parent rollup. Cause: %w", err)
+		return nil, nil, fmt.Errorf("could not fetch rollup receipts. Cause: %w", err)
 	}
 
 	// TODO - #718 - Validate any rollups in the block against the stored batches.
 
-	if err = rc.storage.UpdateL2HeadForL1Block(block.Hash(), currentHeadRollup, nil); err != nil {
```

Comment:
Instead of doing two L2 head updates - one before the rollup is produced, and one once the rollup is produced - we now do a single L2 head update once the rollup is produced.